packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -1034,7 +1034,11 @@ class CupertinoDialogAction extends StatelessWidget {
     this.isDestructiveAction = false,
     this.textStyle,
     @required this.child,
-  }) : assert(child != null);
+  }) : assert(child != null),
+       assert(isDefaultAction != null),
+       assert(isDestructiveAction != null),
+       assert(!(isDefaultAction && isDestructiveAction),
+          "Action can't be default and destructive at the same time");
```

Comment:
On iOS an action can be both default(preferredAction) and destructive.

Reply:
Thanks to point it out, I believed was not possible since it's an enum on iOS.

Reply:
If I'm not wrong on iOS you have en enum for `default`, `cancel` and `destructive` and a boolean for `preferredAction `?. Maybe I can bring the behavior closer to the native one, but it will be a breaking change.

Reply:
Sorry for the ambiguity. The iOS enum case [`default`](https://developer.apple.com/documentation/uikit/uialertaction/style/default) doesn't make the action button bold. `isDefaultAction` is similar to [UIAlertController.preferredaction](https://developer.apple.com/documentation/uikit/uialertcontroller/1620102-preferredaction) which takes a `UIAlertAction?` and highlights it.

I'm filing a new issue for the misleading parameter name.